src/js/utils/range-segments-mixin.js

```diff
@@ -0,0 +1,140 @@
+import { document } from './server-safe-globals.js';
+import { getOrInsertCSSRule, insertCSSRule } from './element-utils.js';
+import { observeResize, unobserveResize } from './resize-observer.js';
+
+const template = document.createElement('template');
+template.innerHTML = /*html*/`
+  <style>
+    .segments #appearance {
+      height: var(--media-range-segment-hover-height, 7px);
+    }
+
+    #background,
+    #track {
+      clip-path: url(#segments-clipping);
+    }
+
+    #segments {
+      --segments-gap: var(--media-range-segments-gap, 2px);
+      position: absolute;
+      width: 100%;
+      height: 100%;
+    }
+
+    #segments-clipping {
+      transform: translateX(calc(var(--segments-gap) / 2));
+    }
+
+    #segments-clipping:empty {
+      display: none;
+    }
+
+    #segments-clipping rect {
+      height: var(--media-range-track-height, 4px);
+      y: calc((var(--media-range-segment-hover-height, 7px) - var(--media-range-track-height, 4px)) / 2);
+      transition: var(--media-range-segment-transition, transform .1s ease-in-out);
+      transform: var(--media-range-segment-transform, scaleY(1));
+      transform-origin: center;
+    }
+  </style>
+  <svg id="segments"><clipPath id="segments-clipping"></clipPath></svg>
+`;
+
+/** @typedef {new (...args: any[]) => any} Constructor */
+/**
+ * @template {!Constructor} T
+ * @param {T} superclass - The class to extend
+ */
+export const RangeSegmentsMixin = (superclass) =>
```

Comment:
this was especially added because of your feedback on making it a more composable feature.
yes, this can be mixed in to any kind of media-chrome-range that requires range segmentation.

super/subclasses don't fit as good here I think.

I think this could be a nice pattern to add some other features like range markers too.
keeps the features more isolated and the template can be dynamically added /removed only when the feature is enabled.

Reply:
Seeing as we don't have a clear second use case for this yet (at least none that I'm hearing), the composability aspect feels premature to me. And the cost is making it harder understand what's happening when you're reading the media-time-range code for the first time.

Maybe if we had a `media-time-range` and a `media-time-range-with-captions`. But I'm not proposing that. Or maybe if the media-time-range code file was getting really big, but I don't think that's the case yet.

All that to say I don't have a problem with shipping this as is. I would just prefer we have clearer reasons for doing this in the future, since there is a cost.

Reply:
ok let me merge this back into media-chrome-range then